tests/js/web/NavMainComponentTest.js

```diff
@@ -233,6 +242,32 @@ describe('NavMainComponent', function () {
         });
     })
 
+    describe('clicking on a nav link with a missing href', function () {
+        beforeEach(function () {
+            this.navMainComponent.init();
+            this.clickEvent = $.Event('click');
+        });
+
+        it('should throw an error', function () {
+            expect(() => {
+                this.$missingHref.trigger(this.clickEvent);
+            }).to.throw('A nav link must have a href or data-target attribute');
+        });
+    });
+
+    describe('clicking on a nav link with a missing data-target', function () {
+        beforeEach(function () {
+            this.navMainComponent.init();
+            this.clickEvent = $.Event('click');
+        });
+
+        it('should throw an error', function () {
+            expect(() => {
+                this.$missingDataTarget.trigger(this.clickEvent);
+            }).to.throw('A nav link must have a href or data-target attribute');
```

Comment:
One super minor thing here, these string constants would be good candidates as static "class" properties. So we could perhaps define them on the `NavMainComponent` like so...

```
NavMainComponent.MISSING_ATTR_ERROR = "You broke it";

// throw this string
throw new Error(NavMainComponent.MISSING_ATTR_ERROR);

// assert this string
...to.throw(NavMainComponent.MISSING_ATTR_ERROR);
```

Reply:
Do you mean define them in the actual component, not the test? If so, where?

Reply:
yes in the component itself, I normally put them at the top 

```
Foo.GREETING = 'Hello, World!';

function Foo () {}

Foo.prototype.sayGreeting = function () {
	console.log(Foo.GREETING);
}
```

They could go anywhere (outside of a method / block) in that file though.